common/beerocks/bcl/source/son/son_wireless_utils.cpp

```diff
@@ -462,49 +496,13 @@ std::vector<uint8_t> wireless_utils::calc_5g_20MHz_subband_channels(
  * @param operating_class operating class
  * @return std::set<uint8_t> set of supported channels by the operating class or empty if failure
  */
-std::set<uint8_t> wireless_utils::operating_class_to_channel_set(uint8_t operating_class)
+const std::set<uint8_t> &wireless_utils::operating_class_to_channel_set(uint8_t operating_class)
 {
-    static const std::map<uint8_t, std::set<uint8_t>> operating_class_to_channel_set = {
-        {81, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13}},
-        {82, {14}},
-        {83, {1, 2, 3, 4, 5, 6, 7, 8, 9}},
-        // TODO channels for operating classes 85,96,87 should be taken from the regulatory domain
-        {94, {133, 137}},
-        {95, {132, 134, 136, 138}},
-        {96, {131, 132, 133, 134, 135, 136, 137, 138}},
-        {101, {21, 25}},
-        {102, {11, 13, 15, 17, 19}},
-        {103, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}},
-        {104, {184, 192}},
-        {105, {188, 196}},
-        {106, {191, 195}},
-        {107, {189, 191, 193, 195, 197}},
-        {108, {188, 189, 190, 191, 192, 193, 194, 195, 196, 197}},
-        {109, {184, 188, 192, 196}},
-        {110, {183, 184, 185, 186, 187, 189}},
-        {111, {182, 183, 184, 185, 186, 187, 189}},
-        {112, {8, 12, 16}},
-        {113, {7, 8, 9, 10, 11}},
-        {114, {6, 7, 8, 9, 10, 11}},
-        {115, {36, 40, 44, 48}},
-        {116, {36, 44}},
-        {117, {40, 48}},
-        {118, {52, 56, 60, 64}},
-        {119, {52, 60}},
-        {120, {56, 64}},
-        {121, {100, 104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 144}},
-        {122, {100, 108, 116, 124, 132, 140}},
-        {123, {104, 112, 120, 128, 136, 144}},
-        {124, {149, 153, 157, 161}},
-        {125, {149, 153, 157, 161, 165, 169}},
-        {126, {149, 157}},
-        {127, {153, 161}},
-        {180, {1, 2, 3, 4, 5, 6}}};
-
-    auto it = operating_class_to_channel_set.find(operating_class);
-    if (it == operating_class_to_channel_set.end()) {
+
+    auto it = operating_classes_list.find(operating_class);
+    if (it == operating_classes_list.end()) {
         LOG(ERROR) << "reserved operating class " << int(operating_class);
-        return std::set<uint8_t>();
+        return operating_classes_list.at(0).channels;
```

Comment:
at(0) will return this - {0, {{}, beerocks::BANDWIDTH_MAX}}};

Reply:
Sorry, I didn't notice this element.

Reply:
It was taken from moran's comment to 459:
https://github.com/prplfoundation/prplMesh/pull/459#discussion_r341846613

Reply:
I think @morantr could have lived with a static dummy variable as well. The main point was to have something static so you can return a const ref instead of a copy of the set.

Anyway, (nitpick)

Reply:
Done.